### PR TITLE
Feature/account server gateway

### PR DESCRIPTION
**Pull request by romeovs**
Adds client to `core/account` to manage gateways on the account server.

### CODE REVIEW

File: core/account/types.go [hunk 1]
```diff
@@ -59,3 +59,22 @@ type Component struct {
...
 }
+
+// Gateway represents a gateway on the account server
+type Gateway struct {
+	ID               string           `json:"id" valid:"required"`
+	EUI              types.GatewayEUI `json:"eui" valid:"required"`
+	Activated        bool             `json:"activated"`
+	FrequencyPlan    string           `json:"frequency_plan"`
+	FrequencyPlanURL string           `json:"frequency_plan_url"`
+	PublicLocation   bool             `json:"location_public"`
```
Comment (htdvisser) [inline core/account/types.go]: Be consistent. `StatusPublic` vs `PublicLocation`.

File: core/account/types.go [hunk 2]
```diff
@@ -59,3 +59,22 @@ type Component struct {
...
+	StatusPublic     bool             `json:"status_public"`
+	Location         *Location        `json:"location"`
+	Collaborators    []Collaborator   `json:"collaborator"`
+	Key              string           `json:"key"`
+}
+
+type Location struct {
+	Lng float64 `json:"lng"`
+	Lat float64 `json:"lat"`
+}
```
Comment (htdvisser) [inline core/account/types.go]: Please don't shorten `Lat` and `Lng` in the struct.

File: core/account/gateways.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+// ChangeFrequencyPlan changes the requency plan of a gateway
+func (a *Account) ChangeFrequencyPlan(gatewayID, plan string) (Gateway, error) {
+	return a.EditGateway(gatewayID, GatewayEdits{
+		FrequencyPlan: plan,
+	})
+}
+
+// ChangeLocation changes teh location of the gateway
+func (a *Account) ChangeLocation(gatewayID string, location Location) (Gateway, error) {
```
Comment (htdvisser) [inline core/account/gateways.go]: Maybe just make this `ChangeLocation(gatewayID string, latitude, longitude float64)` ?

Comment (romeovs) [inline core/account/gateways.go]: The problem there is that its is easy to forget which one comes first, so the API becomes less clear.

Comment (htdvisser) [inline core/account/gateways.go]: Don't people use godoc/gocode/godef anymore?

**Pull request merged**